Enable ExampleValidationModule to support partial examples

StarKhan6368 opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->

**What**: Enable ExampleValidationModule to support partial examples

<!-- Have you done all of these things?  -->

**Checklist**:

<!-- add "N/A" to the end of each line that's irrelevant to your changes -->

<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->

- [x] Unit Tests
- [x] Build passing locally
- [ ] Sonar Quality Gate
- [ ] Security scans don't report any vulnerabilities
- [ ] Documentation added/updated (share link)
- [ ] Sample Project added/updated (share link)
- [ ] Demo video (share link)
- [ ] Article on Website (share link)
- [ ] Roadmpap updated (share link)
- [ ] Conference Talk (share link)

StarKhan6368 committed:
helper methods to extract values from partial

StarKhan6368 committed:
Modify matches to work with partials and 4xx

StarKhan6368 committed:
Wire up ExampleValidationModule properly

- To work with partial example
- Add test for the same

StarKhan6368 committed:
Modify getExistingExamples in ExampleModule

- To work with partials and path mutated 4xx
- Add tests for the same

StarKhan6368 committed:
Merge branch 'main' into validate-cmd-partial-example

joelrosario committed:
Minor refactoring

joelrosario committed:
Merge branch 'validate-cmd-partial-example' of github.com:znsio/specmatic into validate-cmd-partial-example

joelrosario commented on core/src/main/kotlin/io/specmatic/core/Feature.kt:
```diff
@@ -635,21 +625,38 @@ data class Feature(
-        return HasValue(scenarioAsTest(concreteTestScenario, null, Workflow(), originalScenario))
+        val scenario = originalScenario.useExamples(mapOf(apiIdentifier to listOf(example.toRow(specmaticConfig))))
+        return scenario.generateTestScenarios(flagsBased).firstOrNull {
+            it is HasValue && it.value.exampleName == example.testName
```
Need to think through this a little more to ensure that the scenario that we get exactly the concreteTestScenario (filled in if partial), and that flagsBased has no way to influence this.

StarKhan6368 replied:
Updated the logic, let me know if this makes sense

StarKhan6368 committed:
Ensure test matches example after fill-in

- Add test for the same

StarKhan6368 committed:
Add test for attribute selection

StarKhan6368 committed:
Fix attribute selection matching logic

- Add tests for partial and non-partial attr selected examples
- Fix BreadCrumbs in JSONArrayValue

joelrosario committed:
Minor refactoring

joelrosario committed:
Factored out pattern resolution and fillInTheBlanks for HttpRequest in Scenario
